modules/miscutil/lib/upgrades/invenio_2015_01_13_hide_holdings.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2014 CERN.
+#
+# Invenio is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the Free Software Foundation,
+# Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+"""Upgrade for explicitly hiding the 'holdings' tab for all collections\
+ except 'Books' without an existing rule"""
+
+from invenio.dbquery import run_sql
+
+depends_on = ['invenio_release_1_1_0']
+
+
+def info():
+    """Upgrade recipe information."""
+    return "Updates the collectiondetailedrecordpagetabs to hide 'holdings' for every collection \
+            except 'Books' without an existing rule."
+
+
+def do_upgrade():
```

Comment:
@blixhavn Yes, I mentioned that my concern "is perhaps not probable but which can occur".  Imagine a site having 37 collections defined, where `collectiondetailedrecordpagetabs` only specifies values for collections 1, 8, 17, 18, 19.  (The current demo site offers something similar.) Please note that there are no data for collection 2 defined; it is missing from the table.  With your script, you would force Atlantis hard-coded defaults onto collection 2, while with my amendment, you would take existing collection 1 settings, and apply them to all daugher collections that do not have anything defined.  Which I think would be better.  

That's is the difference I was having in mind. An improvement would be to traverse the collection tree in the order of closest parents, and apply that, instead of taking collection 1 settings.  But provided we print some warning for people to check their collection tab settings, we do not really need to do that, I think.